File: src/main/java/ml/duncte123/skybot/commands/musicJava/LeaveCommand.java
/*
 * Skybot, a multipurpose discord bot
 *      Copyright (C) 2017  Duncan "duncte123" Sterken
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package ml.duncte123.skybot.commands.musicJava;

import ml.duncte123.skybot.objects.command.MusicCommand;
import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
import net.dv8tion.jda.core.managers.AudioManager;

public class LeaveCommand extends MusicCommand {

    public final static String help = "make the bot leave your channel.";

    @Override
    public void executeCommand(String invoke, String[] args, GuildMessageReceivedEvent event) {

        AudioManager manager = getAudioManager(event.getGuild());

        if(manager.isConnected()){
            if(!manager.getConnectedChannel().getMembers().contains(event.getMember())){
                sendMsg(event,"I'm sorry, but you have to be in the same channel as me to use any music related commands");
                return;
            }

        }else{
            sendMsg(event,"I'm not in a channel atm");
            return;
        }

        if(manager.isConnected()) {
            getMusicManager(event.getGuild()).player.stopTrack();
            manager.setSendingHandler(null);
            manager.closeAudioConnection();
           sendMsg(event,"Leaving your channel");
        } else {
            sendMsg(event, "I'm not connected to any channels.");
        }


    }

    @Override
    public String help() {
        // TODO Auto-generated method stub
        return help;
    }

    @Override
    public String getName() {
        return "leave";
    }

    @Override
    public String[] getAliases() {
        return new String[]{"disconnect"};
    }
}

File: bot/src/main/kotlin/ml/duncte123/skybot/commands/music/LeaveCommand.kt
/*
 * Skybot, a multipurpose discord bot
 *      Copyright (C) 2017  Duncan "duncte123" Sterken & Ramid "ramidzkh" Khan & Maurice R S "Sanduhr32"
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package ml.duncte123.skybot.commands.music

import me.duncte123.botcommons.messaging.MessageUtils.sendMsg
import me.duncte123.botcommons.messaging.MessageUtils.sendSuccess
import ml.duncte123.skybot.Variables
import ml.duncte123.skybot.objects.command.CommandContext
import ml.duncte123.skybot.objects.command.MusicCommand
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent

class LeaveCommand : MusicCommand() {

    init {
        this.name = "leave"
        this.aliases = arrayOf("disconnect", "exit", "fuckoff")
        this.help = "Makes the bot leave the current voice channel"
        this.cooldown = MUSIC_COOLDOWN
    }

    override fun run(ctx: CommandContext) {
        val event = ctx.event
        val guild = event.guild

        if (!getLavalinkManager().isConnected(guild)) {
            sendMsg(ctx, "I'm not connected to any channels.")
            return
        }

        ctx.audioUtils.removeMusicManager(guild)
        getLavalinkManager().closeConnection(guild)
        guild.audioManager.sendingHandler = null

        sendSuccess(ctx.message)
    }

    override fun handleEvent(event: SlashCommandInteractionEvent, variables: Variables) {
        val guild = event.guild!!

        if (!getLavalinkManager().isConnected(guild)) {
            event.reply("I'm not connected to any channels.").queue()
            return
        }

        variables.audioUtils.removeMusicManager(guild)
        getLavalinkManager().closeConnection(guild)
        guild.audioManager.sendingHandler = null

        event.reply("Left the voice channel.").queue()
    }
}
